ti/instrumentation/instrumentation.go

```diff
@@ -43,28 +46,43 @@ func getTestSelection(ctx context.Context, runner TestRunner, config *api.RunTes
 		if commitErr != nil {
 			log.Infoln("Failed to get reference commit", "error", commitErr)
 			config.RunOnlySelectedTests = false // TI selected all the tests to be run
-			return selection
+			return selection, moduleList
 		}
 		if lastSuccessfulCommitID == "" {
 			log.Infoln("Test Intelligence determined to run all the tests to bootstrap")
 			config.RunOnlySelectedTests = false // TI selected all the tests to be run
-			return selection
+			return selection, moduleList
 		}
 		log.Infoln("Using reference commit: ", lastSuccessfulCommitID)
 		files, err = getChangedFilesPush(ctx, workspace, lastSuccessfulCommitID, tiConfig.GetSha(), log)
 		if err != nil {
 			log.Errorln("Unable to get changed files list. Running all the tests.", "error", err)
 			config.RunOnlySelectedTests = false
-			return selection
+			return selection, moduleList
 		}
 	} else {
 		files, err = getChangedFilesPR(ctx, workspace, log)
 		if err != nil || len(files) == 0 {
 			log.Errorln("Unable to get changed files list for PR. Running all the tests.", "error", err)
 			config.RunOnlySelectedTests = false
-			return selection
+			return selection, moduleList
 		}
 	}
+
+	//check ticonfig params to allow bazel optimization, and get threshold for max file count
+	tiConfigYaml, err := getTiConfig(workspace, fs)
+	if err != nil {
+		log.Infoln("Ti config parsing before selectTests fails")
+	}
+
+	//skip bazel src inspection if optimization in config not selected
+	if tiConfigYaml.Config.BazelOptimization {
+		//add src files listed in java target rules in the BUILD.bazel files if BUILD.bazel is changed
+		files, moduleList, err = addBazelFilesToChangedFiles(ctx, workspace, log, files, tiConfigYaml.Config.BazelFileCount)
```

Comment:
What happens if `err != nil`?

Reply:
Ideally we should login an error and continue the code-flow (that means bazel optimization logic is not applied)